### ISSUE TITLE
Create "Recipes" node in the documentation

### ISSUE THREAD

**Issue by mtomko** [typelevel/cats-effect#3410]
I was recently looking for a way to solve a simple problem ("start a supervised task that outlives an http request") and ended up asking on discord. I got an answer (thanks @armanbilge) but in the ensuing discussion we decided it might be nice to curate a list of example solutions to simple problems like this one and publish it in the documentation.

I am willing to start writing out the documentation node and will include an example solution for the problem I originally had, but will gladly include other simple examples anyone feels like contributing or requesting. I think it'll be nice for the page to launch with more than one example. If nothing else, I can mine our codebase for common patterns.

**Comment by armanbilge** [comment]
Another name idea: "recipes" :)

I bet a lot of stuff in cats-effect-std can be covered in these recipes. I'll try to keep adding as I think of them.
- [ ] "how to atomically update a ref with the result of an effect" aka `AtomicCell` :)

**Comment by samspills** [comment]
I mentioned in the discord, but I coincidentally wrote a bit yesterday about scheduling an update to a `Ref`, using an fs2 `Stream` and combining it with an http4s api+server to also interact with the ref. There's not a lot of explaining but the [post is here](https://www.blerf.ca/words/refs-and-streams-2/) if people want to check it out as a potential example to include :)

**Comment by arosien** [comment]
The examples can be a kind of "reverse-index" of the standard library, as you often start with a problem and are trying to find the components of the solution. Only having the list of standard library components doesn't help a user unless they already know what the components do.

**Comment by mtomko** [comment]
I've made a draft PR with an example. I may work on some other examples but I wanted to start by getting something into a PR state. I agree with the idea of using this as a reverse index, and I think many of the members of the standard library deserve something here.

**Comment by mtomko** [comment]
It's very tempting to make the example just:

```scala
import scala.concurrent.duration._

import cats.effect.{IO, IOApp}
import cats.effect.std.Supervisor
import cats.syntax.all._

object Joke extends IOApp.Simple {

  val run =
    Supervisor[IO](await = true).use { supervisor =>
      for {
        _ <- supervisor.supervise(IO.sleep(100.millis) >> IO.print("MOO!"))
        _ <- IO.println("Q: Knock, knock!")
        _ <- IO.println("A: Who's there?")
        _ <- IO.println("Q: Interrupting cow.") >> IO.sleep(60.millis)
        _ <- "A: Interrupting cow who?".toList.traverse_(IO.print) >> IO.println("")
      } yield ()
    }

}
```

But timing is always a problem.

**Comment by arosien** [comment]
I love the joke, but shouldn't the "A: ..." `print` be broken up into multiple tokens to be `race`d against the "MOO!"?

**Comment by mtomko** [comment]
Yes, I didn't try that hard because I wasn't sure this was the best example. If people prefer to the the more complicated but more practical example, I can do more. I broke up the printing of the A but didn't intersperse any sleeps, which would probably have helped.

Timing is hard in both comedy and programming.

**Comment by mtomko** [comment]
this works pretty reliably on scastie; if people think it's instructive, we can include a silly example and a complicated example

```scala
import scala.concurrent.duration._

import cats.effect.{IO, IOApp}
import cats.effect.std.Supervisor
import cats.syntax.all._

object Joke extends IOApp.Simple {

  val run =
    Supervisor[IO](await = true).use { supervisor =>
      for {
        _ <- supervisor.supervise(IO.sleep(50.millis) >> IO.print("MOO!"))
        _ <- IO.println("Q: Knock, knock!")
        _ <- IO.println("A: Who's there?")
        _ <- IO.println("Q: Interrupting cow.")
        _ <- IO.print("A: Interrupting cow") >> IO.sleep(50.millis) >> IO.println(" who?")
      } yield ()
    }

}

**Comment by mtomko** [comment]
I'm going to switch my PR to non-draft as soon as the build is happy. I think it probably makes sense for people to contribute their own recipes in, rather than having a group branch/PR.

**Comment by mtomko** [comment]
My initial PR has merged into the 3.4.x branch. It should be safe for people to add to it now. I see a couple of good ideas have been posted and I can try to dream up more.
